### PR TITLE
Splits effect_system.dm into multiple files and cleans it up, OOP particle effects and effect systems

### PR DESCRIPTION
**Pull request by uraniummeltdown**
Removed all relative pathing, only uses absolute pathing
Cleaned up excessive spacing and removed var/ in front of proc arguments
Used some ishelpers
Removed the mustard gas smoke effect as it wasn't used anywhere
Removed useless vars and useless procs like effect/delete() which was just a renamed qdel()
Made effect code OOP reducing a lot of copypasta, moved nanofrost smoke to effects_smoke.dm
Moved expl_particles to the effects_system folder and made it a particle effect
Made foam back into an effect (why was it a structure?)
Fade out effect for nanofrost smoke finally works, it was always broken. Also added the fade out to all smokes
Added uses of QDEL_IN, removed spawns, added addtimers, smoke uses process() now
Sleeping smoke uses Sleeping() instead of AdjustSleeping()
Nanofrost smoke welds scrubbers too now
datum/effect/system becomes datum/effect_system
obj/effect/effect becomes obj/effect/particle_effect
Added a file for /obj/effect path so people don't keep searching for it and expecting specific procs and vars
added PASSGRILLE and PASSTABLE to particle effects

:cl:
fix: Particle effects pass over tables and grilles
fix: Nanofrost smoke fading out actually works now
tweak: All smokes now fade out
tweak: Nanofrost smoke can now also weld scrubbers
/:cl:

### CODE REVIEW

**Comment by FalseIncarnate** [comment]
I think there might be a grand total of like 3 things in the code that break if you remove `src.`

I doubt this is one.

**Comment by uraniummeltdown** [comment]
this is a WIP i'm making smoke OOP

**Comment by uraniummeltdown** [comment]
Ready for review @Crazylemon64 @tigercat2000 @Fox-McCloud

File: code/game/objects/effects/effect_system/effects_sparks.dm [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+/datum/effect_system/spark_spread
+	effect_type = /obj/effect/particle_effect/sparks
+
+//////////////////////////////////
+//////SPARKLE FIREWORKS
+/////////////////////////////////
+////////////////////////////
+/obj/effect/particle_effect/sparks/sparkles
+	name = "sparkle"
```
Comment (tigercat2000) [inline code/game/objects/effects/effect_system/effects_sparks.dm]: \~sparkles\~

**Pull request merged**